### ISSUE TITLE
Fehlende Abbildungen im PDF

### ISSUE THREAD

**Issue by Mo-Gul** [progit/progit2-de#113]
Im PDF des Buches werden die Abbildungen

- [x] 12        Referenz: /03-git-branching/sections/nutshell.asc / Lines 65+66
- [x] 15        Referenz: /03-git-branching/sections/nutshell.asc / Lines 116+117
- [x] 16        Referenz: /03-git-branching/sections/nutshell.asc / Lines 139+140
- [x] 24        Referenz: /03-git-branching/sections/basic-branching-and-merging.asc / Lines 167+168
- [x] 46        Referenz: /03-git-branching/sections/rebasing.asc / Lines 166+167
- [x] 90        Referenz: /06-github/sections/2-contributing.asc / Lines 56+57
- [x] 115      Referenz: /06-github/sections/3-maintaining.asc / Lines 75+76
- [x] 145      Referenz: /08-customizing-git/sections/attributes.asc / Lines 204+205

nicht angezeigt.

**Comment by max123kl** [comment]
Es erleichtert die Fehlersuche erheblich, wenn du auf die entsprechende Quelldatei referenzierst.

**Comment by Mo-Gul** [comment]
Ja, das würde es. Aber wie ich schon in einem anderen Kommentar geschrieben habe, möchte ich gerne hauptsächlich erst einmal **Lesen** und nicht korrigieren, weshalb ich auf genauere Links verzichtet habe.

Ich denke, dass wenn der "Fehler" bei einer Abbildung gefunden wurde, vermutlich auch die anderen automatisch behoben sind.

Ich bin nicht sicher, ob du meinst, dass die Bilder **doch** angezeigt werden. Ich beziehe mich auf das PDF des deutschen Buches v2.1.86 und 
https://github.com/progit/progit2-de/releases/download/2.1.87/progit_v2.1.87.pdf

Dort findet sich Abbildung 12 (bzw. deren Beschriftung) auf Seite 69 (PDF-Seite 75). Das Bild dagegen fehlt ...

**Comment by max123kl** [comment]
> Ich bin nicht sicher, ob du meinst, dass die Bilder doch angezeigt werden. Ich beziehe mich auf das PDF des deutschen Buches v2.1.86 ...

Doch ich habe dich richtig verstanden, **ich** erkenne aber in der .asc Datei keinen Fehler warum die verlinkte Grafik _nicht_ angezeigt wird. Wie schon gesagt, ich bin kein Coder. Vielleicht fällt @cbachhuber oder @pastatopf  etwas auf.

**Comment by Mo-Gul** [comment]
Ok, dann sind wir uns einig, sehr schön.

Genau, einen offensichtlichen Fehler gibt es nicht, das sie auf der Webseite angezeigt werden, das hatte ich natürlich auch gesehen. Aber unschön ist es auf jeden Fall, dass diese Bilder nicht im PDF angezeigt werden. Von daher der Report.

**Comment by max123kl** [comment]
@jnavila 
there are some missing figures in the German PDF output only.
I can't figure out the reason why.

In the first article of this thread, I referenced the missing ones.

**Comment by Mo-Gul** [comment]
Hier noch ein Tipp, wie du noch besser die Quelle nennen kannst. (Bei normalem Quellcode ist das viel einfacher, da kann man direkt die Zeilennummern wählen. Aber weil hier das gerenderte Ergebnis der Datei angezeigt wird, muss man einen kleinen Umweg gehen.

1. Gehe auf die Seite der Datei, von der du die Zeilen verlinken möchtest. Für Abbildung 12 ist dies also https://github.com/progit/progit2-de/blob/master/book/03-git-branching/sections/nutshell.asc.
2. Merke dir den letzten Commit-Hash, der oben rechts angezeigt wird (siehe Screenshot am Ende des Kommentars).
In diesem Fall also `84f892975f561b26659fcb7a4bcfb5dbc96e6c80`
3. Suche die Zeilennummern heraus, die du Referenzieren möchtest. (Auf GitHub kannst du dazu z.B. in die "Blame"-Ansicht wechseln.)
4. Ersetze in dem Link aus 1. `master` durch den Hash und füge am Ende die Zeilennummern mit `#Lx` oder für einen Bereich `#Lx-Ly` an, wobei natürlich `x` und `y` durch die Zeilennummern sind, an. Das ergibt dann in diesem Fall `https://github.com/progit/progit2-de/blob/84f892975f561b26659fcb7a4bcfb5dbc96e6c80/book/03-git-branching/sections/nutshell.asc#L64-L65`

und wird als

https://github.com/progit/progit2-de/blob/84f892975f561b26659fcb7a4bcfb5dbc96e6c80/book/03-git-branching/sections/nutshell.asc#L64-L65

angezeigt.

---
<img src="https://user-images.githubusercontent.com/17992976/86355172-8a3c8a00-bc6a-11ea-9613-10feb80a8850.PNG" width="650">

**Comment by jnavila** [comment]
@max123kl 
> there are some missing figures in the German PDF output only.
> I can't figure out the reason why.
> 
> In the first article of this thread, I referenced the missing ones.

You cannot use commas in  the alternative text (the text between brackets), because this is a list of options and the comma would be interpreted as a separator. But you can quote the whole sentence:

``` asciidoc
.Zwei Branches, die auf dieselbe Serie von Commits zeigen
image::images/two-branches.png["Zwei Branches, die auf dieselbe Serie von Commits zeigen"]
```

**Comment by max123kl** [comment]
Thank you, I will fix that!

**Comment by Mo-Gul** [comment]
Done in PR #125.

**Comment by Mo-Gul** [comment]
Macht es vielleicht Sinn, dass **alle** alternativen Texte in den eckigen Klammern in Anführungszeichen gesetzt werden? Falls sich diese mal ändern sollten oder neue Hinzukommen, hätte man praktisch einen Style-Guide durch **alle** anderen und wendet diesen hoffentlich an. Sollte dann bei einem der Fälle oben ebenfalls wieder ein Komma in der deutschen Übersetzung landen, ist die Gefahr deutlich geringer, dass es vergessen wird.

Wenn ihr (ebenfalls) dafür seid, würde ich ein neues Issue dafür aufmachen.

**Comment by max123kl** [comment]
Die Arbeit halte ich für überflüssig. Vor allem, weil der Grund für die Anführungszeichen später nicht mehr nachvollzogen werden kann.
Aber man könnte, für die Erinnerung, einen Wiki-Eintrag (in der Form einer FAQ) generieren und/oder in der Datei [Special_Characters.asc](https://github.com/progit/progit2-de/blob/master/Special_Characters.asc) einen entsprechenden Passus einfügen.

**Comment by Mo-Gul** [comment]
Der Grund wurde doch (praktisch) genannt. Zusammenhalten des optionalen Arguments.
Ich sehe es wie bei LaTeX; dort ist ebenfalls das Komma ein Trenner von Argumenten. Werte *können* dort immer in geschweiften Argumenten angegeben werden. Darauf kann verzichtet werden, sollte der Wert nicht selbst ein Komma enthalten.
Bei LaTeX macht man den Fehler oft nur ein einziges Mal, wenn man häufiger TeXt.

Hier, bei diesem Projekt sieht die Sache aber m.E. anders aus. Die meisten Leute werden wohl nur fixes Schreiben, weil sie Git lernen wollen. Dafür werden sich die meisten Leute nicht besonders die Asciidoctor-Syntax angucken.

Die Anführungszeichen *mögen* (in den meisten Fällen) überflüssig sein, *schaden* tuen sie allerdings nicht und man wäre *immer* auf der sicheren Seite. Wenn du dir dennoch Sorgen machst, dass man den Grund für die Anführungszeichen vergessen wird, hast du die Lösung selbst genannt: Schreibe einen Wiki-Eintrag und/oder einen Passus in die [Special_Characters.asc](https://github.com/progit/progit2-de/blob/master/Special_Characters.asc)-Datei.

**Comment by max123kl** [comment]
Wenn du dir die Arbeit machen willst, halte ich dich nicht davon ab.
Trotzdem bin ich der Meinung, dass sie zum jetzigen Zeitpunkt überflüssig ist. Den Passus in der [Special_Characters.asc](https://github.com/progit/progit2-de/blob/master/Special_Characters.asc) werde ich demnächst vorbereiten und dann hier zur Diskussion stellen.

**Comment by Mo-Gul** [comment]
Wenn ich _regular expressions_ ein wenig besser kennen würde, wäre das bestimmt ziemlich schnell erledigt ;)
Mal sehen, ob mir da was einfällt, wenn ich mal keine Lust auf Lesen habe und trotzdem noch ein wenig Restenergie für solch eine Aufgabe.
Soll ich ein neues Issue dafür aufmachen, damit es nicht vergessen wird? Kannst es mir ja auch zuweisen.

**Comment by max123kl** [comment]
Ist eigentlich egal.
Ich öffne mal dieses Issue wieder, dann könntest du hier weitermachen oder du entscheidest dich für ein neues Issue, dann kannst du dieses hier schließen.
